pkg/gohai/utils/common.go

```diff
@@ -0,0 +1,183 @@
+// This file is licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"runtime"
+)
+
+var ErrNoFieldCollected = errors.New("no field was collected")
+var ErrArgNotStruct = errors.New("argument is not a struct")
+var ErrNotCollectable = fmt.Errorf("cannot be collected on %s %s", runtime.GOOS, runtime.GOARCH)
+var ErrNotExported = errors.New("field not exported by the struct")
+var ErrCannotRender = errors.New("field inner type cannot be rendered")
+var ErrNoValueMethod = errors.New("field doesn't have the expected Value method")
+var ErrNoJSONTag = errors.New("field doesn't have a json tag")
+
+// reflectValueToString converts the given value to a string, and return a boolean indicating
+// whether it succeeded
+//
+// Supported types are int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64, string
+func reflectValueToString(value reflect.Value) (string, bool) {
+	var rendered string
+	switch value.Kind() {
+	// case reflect.Bool:
+	case reflect.Int:
+		rendered = fmt.Sprintf("%d", value.Interface().(int))
+	case reflect.Int8:
+		rendered = fmt.Sprintf("%d", value.Interface().(int8))
+	case reflect.Int16:
+		rendered = fmt.Sprintf("%d", value.Interface().(int16))
+	case reflect.Int32:
+		rendered = fmt.Sprintf("%d", value.Interface().(int32))
+	case reflect.Int64:
+		rendered = fmt.Sprintf("%d", value.Interface().(int64))
+	case reflect.Uint:
+		rendered = fmt.Sprintf("%d", value.Interface().(uint))
+	case reflect.Uint8:
+		rendered = fmt.Sprintf("%d", value.Interface().(uint8))
+	case reflect.Uint16:
+		rendered = fmt.Sprintf("%d", value.Interface().(uint16))
+	case reflect.Uint32:
+		rendered = fmt.Sprintf("%d", value.Interface().(uint32))
+	case reflect.Uint64:
+		rendered = fmt.Sprintf("%d", value.Interface().(uint64))
+	// case reflect.Uintptr:
+	case reflect.Float32:
+		rendered = fmt.Sprintf("%f", value.Interface().(float32))
+	case reflect.Float64:
+		rendered = fmt.Sprintf("%f", value.Interface().(float64))
+	// case reflect.Complex64:
+	// case reflect.Complex128:
+	// case reflect.Array:
+	// case reflect.Chan:
+	// case reflect.Func:
+	// case reflect.Interface:
+	// case reflect.Map:
+	// case reflect.Pointer:
+	// case reflect.Slice:
+	case reflect.String:
+		rendered = value.String()
+	// case reflect.Struct:
+	// case reflect.UnsafePointer:
+	default:
+		return "", false
+	}
+
+	return rendered, true
+}
+
+// fieldIsValue returns whether the field has a Value function with the correct arguments and return types.
+//
+// Since we don't know the specific T, we can't cast to an interface, and reflect doesn't have any way to check
+// a generic type as long as https://github.com/golang/go/issues/54393 is not implemented.
+func fieldIsValue(fieldTy reflect.StructField) (reflect.Method, bool) {
```

Comment:
The current naming reflects partially what the function does (and `Value` can means value type vs reference type). What do you think about something like `getValueMethod` or something similar?

Reply:
Yes `getValueMethod` makes more sense, I initially named it `fieldIsValue` because I was checking way more things before I simplified the code.